### PR TITLE
add ubuntu agent and cdn with lambda@edge resource for public access

### LINKED ISSUE #132: Migrate existing deployment cdk code from opensearch-build repo opensearch-ci
### Is your feature request related to a problem? Please describe

There is a cdk application that generates stack that generates artifact s3 bucket, relevant roles and policies, a cloudfront distribution and lambda@edge to redirect requests to local artifact s3 bucket. The code is currently hosted in https://github.com/opensearch-project/opensearch-build/tree/main/deployment .

In order to make use of above mentioned components in public jenkins setup it makes more sense to implement that functionality in https://github.com/opensearch-project/opensearch-ci repo. 

### Describe the solution you'd like

After migrating the functionality the resources required to expose artifacts to public must be created in the new public jenkins account. 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rishabh6788**
Signed-off-by: Rishabh Singh <sngri@amazon.com>

### Description
This PR adds support for following:
1. Adds new Ubuntu jenkins agent.
2. Adds CDN distribution with lambda@edge along with supporting IAM role and policies to grant public access to build artifacts  

### Issues Resolved
#132 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: lib/buildArtefacts/build-artifacts-permissions.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      ],
+    });
+
+    const openSearchBundlePolicy = new Policy(scope, 'opensearch-bundle-policy', {
+      statements: [
+        new PolicyStatement({
+          actions: [
+            's3:GetObject*',
+            's3:GetBucket*',
+            's3:List*',
```
Comment (gaiksaya) [inline lib/buildArtefacts/build-artifacts-permissions.ts:61]: I think we should not allow listing. @peterzhuamazon  can weigh in here regarding security concerns. If not required can we remove it?

File: lib/buildArtefacts/build-artifacts-permissions.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          ],
+          resources: [`${s3BucketArn}`],
+          effect: Effect.ALLOW,
+        }),
+        new PolicyStatement(
+          {
+            actions: [
+              's3:GetObject*',
+              's3:GetBucket*',
+              's3:List*',
```
Comment (gaiksaya) [inline lib/buildArtefacts/build-artifacts-permissions.ts:71]: Same as above

File: lib/buildArtefacts/artifacts-public-access.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
+/**
```
Comment (gaiksaya) [inline lib/buildArtefacts/artifacts-public-access.ts:1]: Looks like a typo in the file name `buildArtefacts` -> `buildArtifacts`. Might want to change all imports too after this change

File: lib/compute/agent-nodes.ts [hunk 4]
```diff
@@ -61,5 +63,12 @@ export class AgentNodes {
         amiId: 'ami-020c52efb1a60f1ae',
         initScript: 'sudo yum update -y || sudo yum update -y',
       };
+      this.UBUNTU_X64_DOCKER = {
+        workerLabelString: 'Jenkins-Agent-Ubuntu2004-X64-m52xlarge-Docker-Builder',
+        instanceType: 'M52xlarge',
+        remoteUser: 'ubuntu',
+        amiId: 'ami-0f6ceb3b3687a3fba',
```
Comment (gaiksaya) [inline lib/compute/agent-nodes.ts:70]: Is this AMI public?

**Review by gaiksaya — CHANGES_REQUESTED** [review]

File: lib/compute/agent-nodes.ts (hunk 4 above)
Comment (rishabh6788) [reply lib/compute/agent-nodes.ts:70]: yes.

File: lib/buildArtefacts/artifacts-public-access.ts (hunk 3 above)
Comment (rishabh6788) [reply lib/buildArtefacts/artifacts-public-access.ts]: fixed the naming

File: lib/buildArtefacts/build-artifacts-permissions.ts (hunk 1 above)
Comment (rishabh6788) [reply lib/buildArtefacts/build-artifacts-permissions.ts]: I have replicated the same permission as present in current `opensearch-bundle` role in current jenkins account

**Comment by peterzhuamazon** [comment]
@gaiksaya @rishabh6788 I think 
```
's3:GetBucket*',
              's3:List*',
```

These two can be commented out for now, the getObject one is good for me.
Also remember I didnt create these it was Marc and others at the time.
So it is possible they have too much permissions here.

**Comment by peterzhuamazon** [comment]
@gaiksaya @rishabh6788 after further debugging it is better we remove `s3:List*` but add `"s3:ListObjectsV2"`.
This is required for `s3 sync` command in Jenkins.
This should not cause issue in cloudfront as we only set cloudfront to use `s3:getObject` in bucket permissions.

**Comment by rishabh6788** [comment]
@peterzhuamazon @gaiksaya Addressed your concerns regarding S3 bucket access. Please review.

**Review by gaiksaya — APPROVED** [review]

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**